### ISSUE TITLE
Broken compatibility with Ruby 2.3: NoMethodError (undefined method `set_key' for #<OpenSSL::PKey::RSA:0x007f2b5ceb76c8>):

### ISSUE THREAD

**Issue by ProgVal** [tootsuite/mastodon#2029]
Hi,

when getting toots from some users (eg. @Reventlov@mastodon.social), I get a 500 error, with this in the log:

```
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71]
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] NoMethodError (undefined method `set_key' for #<OpenSSL::PKey::RSA:0x007f2b5ceb76c8>):
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71]
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] app/services/follow_remote_account_service.rb:49:in `call'
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] app/services/follow_service.rb:10:in `call'
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] app/controllers/api/v1/accounts_controller.rb:66:in `follow'
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] app/controllers/concerns/localized.rb:24:in `block in set_locale'
Apr 17 21:19:24 ruby bundle[18045]: [c5556bf9-4486-420a-8191-6b9090b86d71] app/controllers/concerns/localized.rb:23:in `set_locale'
```

I use version 1.2, and ran bundler after the update, whose output contained:

```
Installing openssl 2.0.3 with native extensions
```

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by ProgVal** [comment]
It looks like this is caused by me using Ruby 2.3.3, which is marked as supported in the Gemfile since 061120914.

**Comment by zorun** [comment]
This looks really similar to #1670 (in fact, I suspect that the fix for #1670 broke support for ruby 2.3.x)

That being said, I'm running mastodon 1.2 with ruby 2.3.3 and did not see any issue

**Comment by ProgVal** [comment]
Yup https://github.com/tootsuite/ostatus2/pull/2/files#diff-02456f015f11b48d1ed078b786a03755R8
